### ISSUE TITLE
How to control scope of services returned by DiscoveryClient 

### ISSUE THREAD

**Issue by raffian** [spring-cloud/spring-cloud-kubernetes#135]
In our kubernetes non-prod installation, we suffix `-<env>` to project names to organize deployments for applications in those namespaces by their target runtime environment. Is there a way to filter services returned by DiscoveryClient such that an application deployed to `qa` is allowed to discover only services deployed to `qa`? We're using ovs-multitenant platform security, so applications, by default, are isolated within the bounds of their namespace. In our preliminary tests with hello world apps, the list of services returned by DiscoveryClient appear to be limited to those in the same project, consistent with default security policy we currently employ. So is that how it works; the scope of accessible services is determined by project security policy?

inventory-qa
inventory-uat
billing-dev
billing-qa
billing-uat

**Comment by christophetd** [comment]
Looks like it's related to #134. A filter feature would definitely be great to have!

**Comment by christophetd** [comment]
Could be something around the lines of https://gist.github.com/christophetd/971cdc8e330eb237be37e751431f842f

**Comment by christophetd** [comment]
It could also be implemented as a filter, which a feature I suggested in #134.

**Comment by raffian** [comment]
Just to be clear, `billing-dev` is the namespace/project name; applications in that namespace using DiscoveryClient may be SearchApp, FrontEndUI, EmailProcessor. If DiscoveryClient has a method for getting the service's namespace, the suffix should be obtained from that.

**Comment by christophetd** [comment]
Ah, missed it - I thought `billing-dev` et al. were service names. No idea in this case.

**Comment by raffian** [comment]
I'm still not clear how services are filtered by DiscoveryClient. During our first test we got an error; our  application was not able to talk to kubernetes REST Api; we expected that, so we updated the project's security policy as per the documentation:

`oc policy add-role-to-user cluster-reader system:serviceaccount:billing-dev:default`

Once we did that the error disappeared. I suppose the next test we should do is deploy applications in different namespaces, let them register, and see how DiscoveryClient filters the list.
